feat/src/main/java/com/example/feat/document/controller/DocumentViewModel.kt

```diff
@@ -103,10 +107,44 @@ class DocumentViewModel @Inject constructor(
                     tutors = tutors,
                     favoriteTutors = favoriteTutors,
                     isLoading = false,
+                    totalPage = tutorResponse.count
                 )
             }
         } else {
             setUiState { copy(isLoading = false) }
         }
     }
+
+    private fun pageFetchTutorial() {
+        if(uiState.value.isLoading) return
+        val currentPage = uiState.value.currentPage
+        val totalPage = uiState.value.totalPage
+        if((currentPage * TUTOR_LIMIT_ITEM) >= totalPage) return
+        viewModelScope.launch {
+            setUiState { copy(isLoading = true) }
+            if (currentPage < totalPage) {
+                documentUseCase.fetchTutors(
+                    paginationRequest = PaginationRequest(
+                        page = currentPage + 1,
+                        pageSize = TUTOR_LIMIT_ITEM,
+                    )
+                ).collect { either ->
+                    if (either.isRight()) {
+                        val tutors = (either.rightValue() as TutorFavorites).tutors
+                        val favoriteTutors = (either.rightValue() as TutorFavorites).favoriteTutors
```

Comment:
Repeated casting of 'either.rightValue()' should be avoided by storing the result in a variable, as it affects performance and maintainability.

```diff
- val tutors = (either.rightValue() as TutorFavorites).tutors
- val favoriteTutors = (either.rightValue() as TutorFavorites).favoriteTutors
+ val tutorFavorites = either.rightValue() as TutorFavorites
+ val tutors = tutorFavorites.tutors
+ val favoriteTutors = tutorFavorites.favoriteTutors
```